core/src/main/java/hudson/cli/CLIReturnCodeStandard.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2018, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package hudson.cli;
+
+/**
+ * Set of standard CLI return code
+ * @since TODO
+ */
+public enum CLIReturnCodeStandard implements CLIReturnCode {
+	/**
+	 * Everything went ok
+	 * HTTP equivalent: 2xx
+ 	 */
+	OK(0),
```

Comment:
FWIW in propriatarty code I find I am doing similar to this but with the addition of a String so I can 
a) send the reaso to the user
b) log a human understandable message in the Jenkins log.

have you thought about making a user description mandatory here - to actually force the improvement of the experience by making people think about usability?

Reply:
for example  
try {
  //something that can fail
} 
catch (Throwable t) {
  return UNKNOWN_ERROR_OCCURRED(t.getMessage();
}

and the CLI can actually send that to the user (and log it) so that you don't duplicate this everywhere.
infact even if the run or execute method throws an exception the other layer can catch that...

Reply:
I agree, but perhaps directly in the interface CLIReturnCode, instead of the CLIReturnCodeStandard?